src/utils/wallet.js

```diff
@@ -806,6 +806,8 @@ class Wallet {
             signer: new nearApiJs.InMemorySigner(tempKeyStore)
         })
         
+        const connectionConstructor = this.connection
```

Comment:
@marcinbodnar @mattlockyer I think a proper fix would be avoiding modifying `this.connection` in the first place

Reply:
@vgrichina agree, this is just a hotfix from my side, that I think should be merged because it might affect users that will be working with wallet straight after account recovery. @mattlockyer can you handle it? Refactor `getAccount` to use custom connection is easy, but there is also `this` passed to `TwoFactor` which you know the most about.

Reply:
@marcinbodnar yeah, discussing longer term fix with @mattlockyer